app/controllers/refusal_advice_controller.rb

```diff
@@ -3,12 +3,45 @@
 # after their Info Requests are marked as refused/rejected.
 #
 class RefusalAdviceController < ApplicationController
+  before_action :authenticate
+
   def create
-    @params = parsed_refusal_advice_params # TODO: move into log event method
+    log_event
+
+    case parsed_refusal_advice_params.fetch(:button)
+    when 'clarification'
+      redirect_to new_request_followup_path(
+        request_id: info_request.id, anchor: 'followup'
+      )
+
+    when 'internal_review'
+      redirect_to new_request_followup_path(
+        request_id: info_request.id, internal_review: '1', anchor: 'followup'
+      )
+
+      # TODO: there are other actions - define what to do in the refusal advice
+      # YAML?
+    end
   end
 
   private
 
+  def authenticate
+    authenticated?({})
+  end
+
+  def info_request
+    @info_request ||= current_user.info_requests.
```

Comment:
should check what happens with "sign in as" here.

Reply:
Confirmed this isn't an issue